app/src/lib/shells/win32.ts

```diff
@@ -248,6 +285,17 @@ export function launch(
         shell: true,
         cwd: path,
       })
+    case Shell.Cygwin:
+      const cygwinPath = `"${foundShell.path}"`
+      log.info(`launching ${shell} at path: ${cygwinPath}`)
+      return spawn(
+        cygwinPath,
+        [`/bin/sh -lc 'cd "$(cygpath "${path}")"; exec bash`],
```

Comment:
Do we need to pass in all these arguments? I thought emulating the desktop shortcut would be enough here...

Reply:
Unfortunately I didn't find a better way to open Cygwin from a certain path and this is the best solution google search gave me. The --cd flag that works for Git Bash doesn't work for Cygwin and there is just no flag in general in Cygwin for that function.

Reply:
I found there's an unofficially-documented `--dir` flag here: https://github.com/mintty/mintty/wiki/Tips#starting-in-a-particular-directory

This launched me to the repository, mapped under `/cygdrive/`

<img width="761" src="https://user-images.githubusercontent.com/359239/45625348-f7b20300-ba62-11e8-817d-8d3fd8252ef8.png">

```diff
diff --git a/app/src/lib/shells/win32.ts b/app/src/lib/shells/win32.ts
index fe48bb699..df000116c 100644
--- a/app/src/lib/shells/win32.ts
+++ b/app/src/lib/shells/win32.ts
@@ -290,7 +290,7 @@ export function launch(
       log.info(`launching ${shell} at path: ${cygwinPath}`)
       return spawn(
         cygwinPath,
-        [`/bin/sh -lc 'cd "$(cygpath "${path}")"; exec bash`],
+        ['-i', '/Cygwin-Terminal.ico', '--dir', path, '-'],
         {
           shell: true,
           cwd: path,
```

Reply:
Tried changing this to your solution and getting a problem:
![image](https://user-images.githubusercontent.com/43252111/45647336-4cd72e80-bace-11e8-95f8-482a41f969bf.png)
Also I wasn't able to find any information about --dir flag.

Reply:
Try wrapping `path` in quotes like ``"${path}"`` and see if that changes the behaviour (I was slack and didn't have unsafe characters in my account name).

Reply:
I spent a long time figuring out why your method didn't work for folders with spaces (your last comment didn't fix it unfortunately) and I think I get it now.
basically what --dir does from my understanding is takes the path you give, runs the command "cygpath" on it and then runs cd on the result.

Now if we look at the example of my directory, I did some string manipulation to get the path in correct format for cygpath (before the spawn command):
      var newpath = path.replace(/\\/g, '\\\\')
      newpath = newpath.replace(/\s/, '\\ ')

So now the path I get is:
C:\\Users\\Max\ Kolchinsky\\Documents\\max_sbox\\desktop
when I run cygpath on it inside cygwin I get:

$ cygpath C:\\Users\\Max\ Kolchinsky\\Documents\\max_sbox\\desktop
/cygdrive/c/Users/Max Kolchinsky/Documents/max_sbox/desktop

But the desired result for "cd" command to work is
/cygdrive/c/Users/Max\ Kolchinsky/Documents/max_sbox/desktop
(basically we need one more backslash to escape the space for the cd command to work.

So since the cd command fails when I open the shell it doesn't redirect me to the desired folder.

By the way could you please tell me how you paste diff changes inside comments here?
I tried finding a way in the "Mastering Markdown" page but didn't find anything useful.

Reply:
> By the way could you please tell me how you paste diff changes inside comments here?

Check out the "Syntax Highlighting" section here: https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown

Reply:
> In my honest opinion, I think that my previous solution is better because it takes care of this issue regardless of folders that contain spaces.

I'm fine with that, especially if it handles the corner cases

Reply:
Thank you. In that case I propose we keep my change. I also added a commit where I fixed the cygwin 32bit issue you pointed out yestarday.